experiment/simulation/js/animation.js

```
@@ -0,0 +1,672 @@
+    }
+}
+function appendInputCin() {
+    if (TEXTINPUT[2].textContent !== "0" && timeline.progress() === 0) {
+        changeto0(245,115,2,2);
+    }
+    else if (TEXTINPUT[2].textContent !== "1" && timeline.progress() === 0) {
+        changeto1(245,115,2,2);
+    }
+    setter(TEXTINPUT[2].textContent,DOTS[8]);
+}
+
+function appendInputS1() {
+    if (TEXTINPUT[3].textContent !== "0" && timeline.progress() === 0) {
+        changeto0(515,125,3,3);
+    }
+    else if (TEXTINPUT[3].textContent !== "1" && timeline.progress() === 0) {
+        changeto1(515,125,3,3);
+    }
+    setter(TEXTINPUT[3].textContent,DOTS[9]);
+}
+
+function appendInputS2() {
+    if (TEXTINPUT[4].textContent !== "0" && timeline.progress() === 0) {
+        changeto0(545,125,4,4);
+    }
+    else if (TEXTINPUT[4].textContent !== "1" && timeline.progress() === 0) {
+        changeto1(545,125,4,4);
+    }
+    setter(TEXTINPUT[4].textContent,DOTS[10]);
+}
+
+function changeto1(coordinateX,coordinateY,object,textObject) {
+    TEXTINPUT[textObject].textContent = 1;
+    svg.appendChild(TEXTINPUT[textObject]);
+    setCoordinates(coordinateX,coordinateY,TEXTINPUT[textObject]);
+    fillColor(OBJECTS[object],"#29e");
+    clearObservation();
+    objectAppear(TEXTINPUT[textObject]);
+}
+
+function changeto0(coordinateX,coordinateY,object,textObject) {
+    TEXTINPUT[textObject].textContent = 0;
+    svg.appendChild(TEXTINPUT[textObject]);
+    setCoordinates(coordinateX,coordinateY,TEXTINPUT[textObject]);
+    fillColor(OBJECTS[object],"#eeeb22");
+    clearObservation();
+    objectAppear(TEXTINPUT[textObject]);
+}
+
+function computeFullAdder(a,b,cin) {
+    let sum = parseInt(a) +parseInt(b) + parseInt(cin);
+    if (sum === 0) {
+        return ["0","0"];
+    }
+    else if (sum === 1) {
+        return ["1","0"];
+    }
+    else if (sum === 2) {
+        return ["0","1"];   
+    }
+    else if (sum === 3) {
+        return ["1","1"];
+    }
```

Please provide a default case here, which would show some error. Otherwise any bugs will be silently ignored.
Also if you have a conditional with more than 2-3 conditions, it's better to use a switch-case which is more efficient and more readable.

---

changed to switch-case